### DISCUSSION TITLE
Shorten binary names?

### DISCUSSION THREAD

**Discussion by gavanderhoorn** [Yaskawa-Global/motoros2#107] [Ideas]
We currently use names such as `motoros2_yrc1000_galactic.out`.

While clear, those names are long, and may not work on all controllers we (want) to support. It looks like at least the DX2 has trouble loading such binaries (https://github.com/Yaskawa-Global/motoros2/pull/49#issuecomment-1585010919), and FS is also expected not to be compatible.

I'd suggest we consider shortening the names, such that it's still clear what the binary is for (application name), and what it supports (controller series and ROS 2 version).

Within ROS 2 this is common practice: all ROS 2 versions have a *codename*, which is a single word identifier which is often used instead of the full name. [Humble Hawksbill](https://docs.ros.org/en/rolling/Releases/Release-Humble-Hawksbill.html) is referred to as `humble` for instance.

ROS 2 CI and build jobs also use a scheme to abbreviate the names: [Hbin_ujv8_uJv8](https://build.ros2.org/view/Hbin_ujv8_uJv8/) shows all the `H`umble `bin`ary jobs targetting `u`buntu `j`azzy running on ARM `v8` (ie: `arm64`).

This works because ROS 2 release and code names are all named alphabetically and uniquely, so the first character of a codename can be used to sort entries and is 'guaranteed' to be unique (in quotes, as there are only 26 letters in the alphabet of course).

I'm not proposing we obfuscate the names *that* much of course.

But what about `mr2_yrc_g.out`? That would be `M`oto`R`OS`2` for `YRC`1000 supporting `G`alactic.

And `mr2_dx2_h.out` would be `M`oto`R`OS`2` for `DX2`00 supporting `H`umble.

And `mr2_fs_i.out` would be `M`oto`R`OS`2` for `FS`100 supporting `I`ron.

Note that the names of the distribution `.zip`s can stay as they are, so there the unabbreviated names would still be used. This should help avoid confusion.

**Comment by ted-miller** [comment]
I'm ok with your proposal. But we should add a `1` after `yrc`. (mr2_yrc1_h.out)

Then there's also the question of whether the micro controller uses `m` or `u`.

**Reply by gavanderhoorn** [reply]
> But we should add a `1` after `yrc`. (mr2_yrc1_h.out)

Ok. I had considered that, but wasn't sure it'd be necessary.

> Then there's also the question of whether the micro controller uses `m` or `u`.

I'll leave that up to you. Both are used currently.

We could use `m`, which would be similar to the `.yrcmLib` extension.

We could also use `u`, as that's in the name of the platform `#define` and the VS solution.

**Comment by gavanderhoorn** [comment]
See #109 for a PR implementing the above proposal.
